inference-engine/src/vpu/graph_transformer/src/model/model.cpp

```diff
@@ -1998,6 +2000,7 @@ void ModelObj::reorderStages(
 void ModelObj::setStagesOrder(const Stage& parent, const Stage& child) {
     ++parent->_nextStages[child];
     ++child->_prevStages[parent];
+    _initialStages.erase(child);
```

Comment:
You also need to change `removeStageOrder` method in this case. Because if you remove this order and the stage again can be executed as initial (it has no dependencies and producers), so it should be added to `_initialStages` again. And also take a look at methods that use `setStageOrder`, I believe most of them already call this method, so we can remove it in the caller methods.